docs/enhancements/1-appliedwork/README.md

```diff
@@ -0,0 +1,158 @@
+The AppliedWork CR is to be an anchor on the managed cluster to ease garbage collection of applied workload.
+
+By leveraging this AppliedWork API:
+- Provide a view on the managed cluster the list of work has been applied.
+- Provide the source data for the work agent to delete resources on the managedcluster based on the appied work.
+
+The AppliedWork API is not meant for external use and it's only meant for internal use by the work controller and/or agent only.
+
+### Goals
+
+- Define a new namespace scope AppliedWork internal API.
+
+### Non-Goals
+
+- Implement managed cluster namespace scope workload garbage collection based on AppliedWork.
+- Handle cluster scope workloads.
+
+## Proposal
+
+We propose to introduce a new `AppliedWork` API 
+
+### Design Details
+
+#### API change  
+
+Add a new `AppliedWork` internal API  
+
+```go
+// AppliedWork represents an applied work on managed cluster that is placed
+// on a managed cluster. An appliedwork links to a work on a hub recording resources
+// deployed in the managed cluster.
+// When the agent is removed from managed cluster, cluster-admin on managed cluster
+// can delete the appliedwork to remove resources deployed by the agent.
+// The name of the appliedwork must be the same as {work name}
+// The namespace of the appliedwork should be the same as the resource applied on
+// the managed cluster.
+type AppliedWork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the desired configuration of AppliedWork.
+	Spec AppliedWorkSpec `json:"spec"`
+
+	// Status represents the current status of AppliedWork.
+	Status AppliedtWorkStatus `json:"status,omitempty"`
+}
+
+// AppliedWorkList contains a list of AppliedWork
+type AppliedWorkList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// List of works.
+	Items []AppliedWork `json:"items"`
+}
+
+// AppliedWorkSpec represents the desired configuration of AppliedWork
+type AppliedWorkSpec struct {
+	// WorkName represents the name of the related work on the hub.
+	WorkName string `json:"workName"`
+}
+
+// AppliedtWorkStatus represents the current status of AppliedWork
+type AppliedtWorkStatus struct {
+	// AppliedResources represents a list of resources defined within the work that are applied.
+	// Only resources with valid GroupVersionResource, namespace, and name are suitable.
+	// An item in this slice is deleted when there is no mapped manifest in work.Spec or by finalizer.
+	// The resource relating to the item will also be removed from managed cluster.
+	// The deleted resource may still be present until the finalizers for that resource are finished.
+	// However, the resource will not be undeleted, so it can be removed from this list and eventual consistency is preserved.
+	AppliedResources []AppliedManifestResourceMeta `json:"appliedResources,omitempty"`
```

Comment:
I think we may not need to track appliedResource if it only treats ns scoped resource. Since we can use owneref to clean them.